### PR TITLE
make doctests deterministic

### PR DESCRIPTION
**Pull request by chaudum**

### CODE REVIEW

File: blackbox/docs/src/doc_tests/tests.py [hunk 1]
```diff
@@ -445,101 +417,76 @@ def test_suite():
...
+                       'general/ddl/partitioned-tables.rst',
+                       'general/builtins/aggregation.rst',
+                       'general/builtins/arithmetic.rst',
+                       'general/builtins/scalar.rst',
+                       'general/builtins/table-functions.rst',
+                       'admin/system-information.rst',
+                       'general/dql/selects.rst',
+                       'interfaces/postgres.rst',):
+        s = docsuite(fn,
+                     setUp=setUpLocationsAndQuotes)
```
Comment (mxm) [inline blackbox/docs/src/doc_tests/tests.py]: Would be nice if we would specify the required tables in the test document itself.

Comment (chaudum) [reply blackbox/docs/src/doc_tests/tests.py]: I agree, a more granular table dependency definition would be nice.
Would be another PR.

File: blackbox/docs/src/doc_tests/tests.py [hunk 2]
```diff
@@ -431,115 +350,162 @@ def setUpTutorials(test):
...
-    for fn in ('admin/snapshots.rst',):
-        s = docsuite('../../' + fn,
-                     parser=crash_parser,
-                     setUp=setUpLocationsAndQuotes,
-                     tearDown=tearDownLocationsAndQuotes)
+        tests.append(s)
+
+    for fn in doctest_files('general/dml.rst',):
+        s = docsuite(fn,
+                     setUp=setUpLocationsQuotesAndUserVisits)
```
Comment (mfussenegger) [inline blackbox/docs/src/doc_tests/tests.py]: These would all fit into 1 line =)

File: blackbox/docs/src/doc_tests/tests.py [hunk 3]
```diff
@@ -431,115 +350,162 @@ def setUpTutorials(test):
...
+        tests.append(s)
+
+    for fn in doctest_files('general/dql/union.rst',):
+        s = docsuite(fn,
+                     setUp=setUpPhotosAndCountries)
         s.layer = crate_layer
-        docs_suite.addTest(s)
+        tests.append(s)
+
+    random.shuffle(tests)
```
Comment (mfussenegger) [inline blackbox/docs/src/doc_tests/tests.py]: Maybe add a short comment that we do this to avoid the tests becoming dependent on each other

**Pull request merged**